### PR TITLE
added tests and rakefile

### PR DESCRIPTION
**Pull request by Danylo056**

### CODE REVIEW

File: spec/test_spec.rb [hunk 1]
```diff
@@ -1 +1,94 @@
+require 'fileutils'
+require 'pry'
```
Comment (dimahadgi) [inline spec/test_spec.rb]: remove this before merging to master(since this is only for debug purposes)

File: spec/test_spec.rb [hunk 2]
```diff
@@ -1 +1,94 @@
...
+  app = Rake.application
+  app.load_rakefile
+
+  before(:all)do
+    app['create'].invoke
+  end
+
+  after(:all)do
+    app['delete'].invoke
+  end
```
Comment (dimahadgi) [inline spec/test_spec.rb]: please re-write it using functions. 
For now we are not going to use Rake tasks

File: spec/test_spec.rb [hunk 3]
```diff
@@ -1 +1,94 @@
...
+  it 'REGEX' do
+    spn = spawn('ping localhost', :out=>['C:\test_folder\temp_file.txt', "w+"])
+    Process.wait spn
+    #success?
+    file = File.open('C:\test_folder\temp_file.txt')
+    file_data = file.read.strip
+    expect(/Minimum = \d+ms, Maximum = \d+ms, Average = \d+ms/.match?(file_data)).to be(true)
+    file.close
+  end
+
```
Comment (dimahadgi) [inline spec/test_spec.rb:90]: do not forget to remove empty lines

File: spec/test_spec.rb [hunk 4]
```diff
@@ -1 +1,94 @@
+require 'fileutils'
+require 'pry'
+require 'rake'
 
+RSpec.describe "1" do
```
Comment (dimahadgi) [inline spec/test_spec.rb]: Start using meaning full names instead of "1"

**Review by dimahadgi — CHANGES_REQUESTED** [review]

File: spec/test_spec.rb (hunk 1 above)
Comment (Danylo056) [reply spec/test_spec.rb]: done

File: functions/test_functions.rb [hunk 5]
```diff
@@ -0,0 +1,14 @@
+def create_temp_file(path)
```
Comment (dimahadgi) [inline functions/test_functions.rb:7]: Add docs

File: functions/test_functions.rb [hunk 6]
```diff
@@ -0,0 +1,14 @@
...
+
+  unless File.directory?(dir)
+    FileUtils.mkdir_p(dir)
+  end
+
+  a = File.new(path, 'w+')
+  a.close
+end
+
+def delete_temp_file(path)
```
Comment (dimahadgi) [inline functions/test_functions.rb:23]: Add docs

File: spec/test_spec.rb (hunk 4 above)
Comment (Danylo056) [reply spec/test_spec.rb]: done

File: spec/test_spec.rb (hunk 3 above)
Comment (Danylo056) [reply spec/test_spec.rb:90]: done

**Review by dimahadgi — APPROVED** [review]
Please add docs to functions

File: functions/test_functions.rb (hunk 5 above)
Comment (Danylo056) [reply functions/test_functions.rb:7]: Done.

File: functions/test_functions.rb (hunk 6 above)
Comment (Danylo056) [reply functions/test_functions.rb:23]: Done.

**Pull request merged**